src/content/docs/en/guides/middleware.mdx

```diff
@@ -11,6 +11,11 @@ This also allows you to set and share request-specific information across endpoi
 
 Middleware is available in both SSG and SSR Astro projects.
 
+:::note
+In SSG, the middleware runs during the **build process**. Once the app is built, the middleware won't run in the deployed website. This is valid also for prerendered pages.
+In SSR, the middleware runs once the website is deployed.
```

Comment:
```suggestion
On pre-rendered pages, the middleware runs only during the **build process** to create static pages and does not run in the deployed website. This includes all pages in a `static` (SSG) project.

For on-demand rendered pages in `server` or `hybrid` mode, the middleware runs in the deployed website when the route is requested.
```
What do you think about this? I tried to split up prerendered vs. on-demand. rendered because SSR can have 2 kinds of pages.

Reply:
I think the `hybrid` requires an additional clarification. Astro pages are rendered statically by default unless a user sets `prerender = false`.

Reply:
Thanks @ematipico, I updated to be even more explicit. What do you think now? Too much?  :smile:

Reply:
That's great! Thank you